### PR TITLE
Home work 18

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/main/java/main/Main.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        try {
+            long userId = dbService.addUser("tully2");
+            System.out.println("Added user id: " + userId);
+
+            UsersDataSet dataSet = dbService.getUser(userId);
+            System.out.println("User data set: " + dataSet);
+
+        } catch (DBException e) {
+            e.printStackTrace();
```
Comment (SevostyanovGerman) [inline src/main/java/main/Main.java:37]: check last work in your group, u have the same problems

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/main/java/dbService/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.dbService.dataSets.User;
+import java.util.Optional;
+
+public class AccountService {
+    
+    private static DataBaseService dbService;
+    static {
+        dbService = new DataBaseService();
+        System.out.println("AccountService loaded");
```
Comment (Sofasmile) [inline src/main/java/dbService/AccountService.java]: do we need sout here?

File: src/main/java/dbService/DataBaseService.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ * @author v.chibrikov
+ * <p>
+ * Пример кода для курса на https://stepic.org/
+ * <p>
+ * Описание курса и лицензия: https://github.com/vitaly-chibrikov/stepic_java_webserver
+ */
+public class DataBaseService {
+    private static final String hibernate_show_sql = "true";
+    private static final String hibernate_hbm2ddl_auto = "update"; // create
```
Comment (Sofasmile) [inline src/main/java/dbService/DataBaseService.java]: to we need this comment?

File: src/main/java/dbService/DataBaseService.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+/**
+ * @author v.chibrikov
+ * <p>
+ * Пример кода для курса на https://stepic.org/
+ * <p>
+ * Описание курса и лицензия: https://github.com/vitaly-chibrikov/stepic_java_webserver
+ */
+public class DataBaseService {
+    private static final String hibernate_show_sql = "true";
```
Comment (Sofasmile) [inline src/main/java/dbService/DataBaseService.java]:
```suggestion
    private static final String HIBERNATE_SHOW_SQL = "true";
```

File: src/main/java/dbService/DataBaseService.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        if (!sessionFactory.isClosed()) {
+            sessionFactory.close();
+        }
+        super.finalize();
+    }
+
+    public void close() {
+        sessionFactory.close();
+    }
+
```
Comment (Sofasmile) [inline src/main/java/dbService/DataBaseService.java]: cs

File: src/main/java/dbService/dao/AccountDAO.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    public AccountDAO(Session session) {
+        this.session = session;
+    }
+
+    public Long addUser(String login, String password) {
+        return (Long) session.save(new User(login, password));
+    }
+
+    public User getUserByLogin(String login) {
+        Query query = session.createQuery("from User where login = :paramLogin");
```
Comment (Sofasmile) [inline src/main/java/dbService/dao/AccountDAO.java]:
```suggestion
        Query query = session.createQuery("SELECT FROM User WHERE login = :paramLogin");
```

File: src/main/java/dbService/dataSets/User.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "users")
+public class User implements Serializable {
+    private static final long serialVersionUID = 1235469543;
```
Comment (Sofasmile) [inline src/main/java/dbService/dataSets/User.java:14]: do we need this field here?

File: src/main/java/dbService/dataSets/User.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
Comment (Sofasmile) [inline src/main/java/dbService/dataSets/User.java]: better use wrapper

File: src/main/java/dbService/dataSets/UsersDataSet.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+ * @author v.chibrikov
+ *         <p>
+ *         Пример кода для курса на https://stepic.org/
+ *         <p>
+ *         Описание курса и лицензия: https://github.com/vitaly-chibrikov/stepic_java_webserver
+ */
+@Entity
+@Table(name = "users_lesson")
+public class UsersDataSet implements Serializable {
+    private static final long serialVersionUID = -3581752392903938614L; // Serializable Important to Hibernate!
```
Comment (Sofasmile) [inline src/main/java/dbService/dataSets/UsersDataSet.java]: the same

File: src/main/java/dbService/dataSets/UsersDataSet.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
Comment (Sofasmile) [inline src/main/java/dbService/dataSets/UsersDataSet.java]: wrapper

File: src/main/java/dbService/dataSets/UsersDataSet.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "UserDataSet{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                '}';
+    }
+}
```
Comment (Sofasmile) [inline src/main/java/dbService/dataSets/UsersDataSet.java]: add a blank line

File: src/main/java/dbService/DataBaseService.java [hunk 12]
```diff
@@ -21,8 +21,8 @@
  * Описание курса и лицензия: https://github.com/vitaly-chibrikov/stepic_java_webserver
  */
 public class DataBaseService {
-    private static final String hibernate_show_sql = "true";
-    private static final String hibernate_hbm2ddl_auto = "update"; // create
+    private static final String HIBERNATE_SHOW_SQL = "true";
+    private static final String hibernate_hbm2ddl_auto = "update";
```
Comment (Sofasmile) [inline src/main/java/dbService/DataBaseService.java:25]: UPER_CASE

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**